### PR TITLE
Ana Lisa Sutherland's Hotel : Octos

### PR DESCRIPTION
**Pull request by The-Beez-Kneez**
# Hotel
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? | 

** I had to consider early on in the initial design process whether or not I would have multiple classes and how to deal with the information provided by room. I had three options with room: 1) To make it its own class, 2) To make it information included as a hash/array in Reservation class, or 3) Create a Hotel Module and set the room information to a constant within it.  I ultimately chose option (3) as I wanted it to be utilized by my other 2 classes. **

Describe a concept that you gained more clarity on as you worked on this assignment. | 

** I gained an additional clarity regarding overlapping reservations/dates once I began trying to organize the logic and meet the test requirements for it.**

Describe a _nominal test_ that you wrote for this assignment. | 

**I wanted to be sure that any room ID could be assigned to a reservation.**

Describe an _edge case test_ that you wrote for this assignment. | 

** I wanted to ensure that the overlap? method did not just account for fully contained or containing conflicts, I tested for all either dates (check in or check out) falling on or within existing reservations. Created 9 tests just for this logic.**

How do you feel you did in writing pseudocode first, then writing the tests and then the code? | 
** I again, had issues getting started with the pseudocode portion of the project, I spent a good 1-2 days on just design and class interactions, however. That did make it fairly easy to write my tests and code, once I had pseudocode down. The time spent designing was well worth it.

PS: 
Can someone please double check my indentations. Atom kept telling me some weird things needed ends and I was unsure why (as was Kai). In the end, it ran as expected based on the indentation guiding given to me by command+i. But I am still unsure as to why it needed some ends in certain places and left them off elsewhere. I was home over the weekend puzzling over this.

### CODE REVIEW

**Comment by analisasutherland** [comment]
PS:
Can someone please double check my indentations. Atom kept telling me some weird things needed ends and I was unsure why (as was Kai). In the end, it ran as expected based on the indentation guiding given to me by command+i. But I am still unsure as to why it needed some ends in certain places and left them off elsewhere. I was home over the weekend puzzling over this.

File: lib/admin.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      room_id = nil
+      # Checking if there are no previous reservations
+      if @reservations.length == 0
+        room_id = ROOM_LIST.sample
+        return room_id
+      else
+        # checking if there are no rooms available
+        @reservations.each do |reservation|
+          reservation.overlap?(start_date, end_date)
+          if false
```
Comment (kariabancroft) [inline lib/admin.rb]: You'll need to store the result of `overlap?` in a variable and compare it - this condition will never run.

**Comment by kariabancroft** [comment]
# Hotel
## What We're Looking For

Feature | Feedback
---     | ---
**Design** |
Demonstrated classes having a single responsibility | Yes - The design works for what you have written so far
Demonstrated loose coupling | Mostly
Methods demonstrate a good use of encapsulation, inputs and outputs | Yes
Wave 1 requirements | Yes
Wave 2 requirements | No
Wave 3 requirements | No

You should delete the files outside of the lib and spec folder before submitting the project, it looks right now like you have duplicates of all of them.
